jcstress-samples/src/main/java/org/openjdk/jcstress/samples/high/race/condition/RaceCondition_01_ReadModifyWriteSequence.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package org.openjdk.jcstress.samples.high.race.condition;
+
+import org.openjdk.jcstress.annotations.*;
+import org.openjdk.jcstress.infra.results.III_Result;
+
+import static org.openjdk.jcstress.annotations.Expect.*;
+
+/*
+    How to run this test:
+        $ java -jar jcstress-samples/target/jcstress.jar -t RaceCondition_01_ReadModifyWriteSequence
+ */
+
+/**
+ * This sample demonstrates you how a read-modify-write sequence can lead to surprising results.
+ */
+@JCStressTest
+@Outcome(id = {"150, 100, 150"}, expect = ACCEPTABLE, desc = "2: v=200, 2: v2=100, 2: v=v2, 1: v=100, 1: v1=150, 1: v=v1")
+@Outcome(id = {"250, 150, 150"}, expect = ACCEPTABLE, desc = "1: v=200, 1: v1=250, 1: v=v1, 2: v=250, 2: v2=150, 2: v=v2")
+@Outcome(id = {"250, 100, 250"}, expect = ACCEPTABLE_INTERESTING, desc = "1: v=200, 1: v1=250, 2: v=200, 2: v2=100, 2: v=v2, 1: v=v1")
+@Outcome(id = {"250, 100, 100"}, expect = ACCEPTABLE_INTERESTING, desc = "1: v=200, 1: v1=250, 2: v=200, 2: v2=100, 1: v=v1, 2: v=v2")
+@State
+public class RaceCondition_01_ReadModifyWriteSequence {
+    private volatile int v = 200;
+
+    @Actor
+    public void actor1(III_Result r) {
+        int v1 = v;
+        v1 += 50;
+        v = v1;
+
+        r.r1 = v1;
+    }
+
+    @Actor
+    public void actor2(III_Result r) {
+        int v2 = v;
+        v2 -= 100;
+        v = v2;
+
+        r.r2 = v2;
+    }
```

Comment:
```suggestion
        int t = v;
        t -= 100;
        v = t;
        r.r2 = t;
    }
```

Reply:
Can we call them t1 and t2 for the descriptions in the outcomes, please?
@vlsi asked me for traces in the descriptions:
```
@Outcome(id = {"150, 100, 150"}, expect = ACCEPTABLE, desc = "2: v=200, 2: v2=100, 2: v=v2, 1: v=100, 1: v1=150, 1: v=v1")
@Outcome(id = {"250, 150, 150"}, expect = ACCEPTABLE, desc = "1: v=200, 1: v1=250, 1: v=v1, 2: v=250, 2: v2=150, 2: v=v2")
@Outcome(id = {"250, 100, 250"}, expect = ACCEPTABLE_INTERESTING, desc = "1: v=200, 1: v1=250, 2: v=200, 2: v2=100, 2: v=v2, 1: v=v1")
@Outcome(id = {"250, 100, 100"}, expect = ACCEPTABLE_INTERESTING, desc = "1: v=200, 1: v1=250, 2: v=200, 2: v2=100, 1: v=v1, 2: v=v2")
```

Reply:
Yes, sure.

Reply:
Ok, thanks.

Reply:
Done

Reply:
Can we resolve that?

Reply:
Frankly speaking, I think `1: v=200, 1: t1=250, 2: v=200, 2: t2=100, 2: v=t2, 1: v=t1` is too complicated.

Do you think `200` is important for the test case?
Would it be better to use `0` starting point?
What if only `+` was used?

For instance: "actor 1 uses 10", and "actor2 uses 20". It would be easier than "actor1 uses 100 and actor2 uses 50" because 1-100; 2-50 is harder than 1-10; 2-20;

suppose the actors only add values to the accumulator, then

Boring results would be:
`10, 30, 30: actor1: 0+10; actor2: 10+20`
`30, 20, 30: actor2: 0+20; actor1: 20+10`

interesting results would be
`10, 20, 10; actor1: 0+10; actor2: 0+20; arbiter observes actor1 result`
`10, 20, 20; actor1: 0+10; actor2: 0+20; arbiter observes actor2 result`

Reply:
Let's just move on. I prefer this description -- https://github.com/openjdk/jcstress/pull/88#discussion_r663076870 -- and that's it.

Reply:
Accepted and committed. Thank you, @shipilev for your final proposal!

Just some last remarks for @vlsi:
This read-modify-write sequence reminded me of the popular example of a bank account where one spouse deposits some money (+50) while the other spouse withdraws money (-100). Therefore, I chose 200.
I must confess that I consider your proposal as confusing to me as my proposal was confusing to you ;-)
However, I think it was good that we tried all that out: If something is confusing to you or to me, then it might be confusing for other users as well.

Reply:
Please take it easy, @vlsi. My proposal also lost! However, I don't mind.

Your feedback has helped me a lot and most of your proposals found their way into that code.
Thank you for that, @vlsi!